### PR TITLE
CephFS - Tier2-CEPH-83575825

### PR DESCRIPTION
**Pull request by hkadam134**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by hkadam134** [comment]
Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GV56RS/

File: tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        )
+        if compare_file_attributes(src_nfs_file_attributes, des_nfs_file_attributes):
+            log.info(
+                "File Attributes on a NFS File are Retained after copying to local filesystem"
+            )
+        else:
+            log.info(
+                "File Attributes are not retained on a NFS File after copying to local filesystem"
+            )
+
```
Comment (sumabai) [inline tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py:153]: The next code is mostly repititive, can we add two set of src and dst file paths as array of arrays and iter with for each set, if possible?

File: tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+
+        def clean_up_directory(client, directory_path):
+            client.exec_command(sudo=True, cmd=f"rm -rf {directory_path}")
+
+        clean_up_directory(client1, f"{nfs_mounting_dir}{nfs_dir}")
+        clean_up_directory(client1, f"{local_fs_path}{nfs_dir}")
+
```
[the remark this answers is not in the document]
Comment (hkadam134) [reply tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py]: Updated

File: tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        # Verify Data Integrity using MD5  of both the NFS file and the local file
+        def calculate_md5(client, file_path):
+            return client.exec_command(
+                sudo=True, cmd=f"md5sum {file_path} | awk '{{print $1}}'"
+            )
+
+        nfs_hash = calculate_md5(client1, nfs_file_path)
+        local_hash = calculate_md5(client1, local_file_path)
+        if nfs_hash == local_hash:
```
[the remark this answers is not in the document]
Comment (hkadam134) [reply tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py]: Updated

File: tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py (hunk 1 above)
Comment (hkadam134) [reply tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py]: Done

**Comment by hkadam134** [comment]
Latest Logs after the changes - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-X0B6NN/

**Review by AmarnatReddy — APPROVED** [review]

**Pull request merged**